src/esp/gfx/replay/PlayerCallbacks.h

```diff
@@ -0,0 +1,71 @@
+#ifndef ESP_GFX_REPLAY_PLAYERCALLBACKS_H_
+#define ESP_GFX_REPLAY_PLAYERCALLBACKS_H_
+
+#include "Magnum/Magnum.h"
+#include "esp/assets/Asset.h"
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+
+namespace esp {
+namespace gfx {
+namespace replay {
+
+// Represents either a scene::SceneNode* or batch renderer entity index.
+class GfxReplayNode;
+
+/**
+ * @brief Callbacks that contain the implementation for gfx::replay::Player.
+ */
+struct PlayerCallbacks {
+  using LoadAndCreateRenderAssetInstance = std::function<GfxReplayNode*(
+      const esp::assets::AssetInfo&,
+      const esp::assets::RenderAssetInstanceCreationInfo&)>;
+
+  using ChangeLightSetup = std::function<void(const gfx::LightSetup& lights)>;
+
+  using DeleteAssetInstance = std::function<void(GfxReplayNode*)>;
+
+  using SetNodeTransform =
+      std::function<void(GfxReplayNode*, Magnum::Vector3, Magnum::Quaternion)>;
+
+  using SetNodeSemanticId = std::function<void(GfxReplayNode*, int)>;
+
+  LoadAndCreateRenderAssetInstance loadAndCreateRenderInstance_;
+  DeleteAssetInstance deleteAssetInstance_;
+  SetNodeTransform setNodeTransform_;
+  SetNodeSemanticId setNodeSemanticId_;
+  ChangeLightSetup changeLightSetup_;
+};
+
+/**
+ * @brief Factory method that instantiates PlayerCallbacks boilerplate to handle
+ * scene graph nodes.
+ */
+static inline PlayerCallbacks createSceneGraphPlayerCallbacks() {
```

Comment:
The `static` isn't doing what you probably think it's doing -- it's the same as wrapping the function in a `namespace {`, makes a local copy of the function inside every file that includes it.

I'd actually make it deinlined in `PlayerCallbacks.cpp` instead, that way the header doesn't need to include basically anything, just a bunch of forward declarations for stuff from `gfx::` and `esp::assets::` and `Magnum/Magnum.h` for Magnum forward declarations.